components/skus/browser/rs/lib/src/sdk/orders.rs

```diff
@@ -206,8 +207,41 @@ where
         Ok(order)
     }
 
+    #[instrument]
+    // submit_receipt allows for order proof of payment
+    pub async fn submit_receipt(&self, order_id: &str, receipt: &str) -> Result<(), InternalError> {
+        event!(Level::DEBUG, order_id = order_id, "submit_receipt called");
+        let request_with_retries = FutureRetry::new(
+            || async {
+                let mut builder = http::Request::builder();
+                builder.method("POST");
+                builder.uri(format!("{}/v1/orders/{}/submit-receipt", self.base_url, order_id));
+
+                let receipt_bytes: Vec<u8> = receipt.as_bytes().to_vec();
+                let req = builder.body(receipt_bytes).unwrap();
+                let resp = self.fetch(req).await?;
+                event!(
+                    Level::DEBUG,
+                    response = str::from_utf8(resp.body()).unwrap(),
```

Comment:
`unwrap_or("<invalid body>")`

It looks like the body is supposed to be json; `fetch` actually attempts deserialisation for logging, but doesn't propagate the result.

Reply:
https://github.com/brave/brave-core/pull/13820/commits/448432a1248d454e47559d00032f42420812f015